olive/passes/olive_pass.py

```diff
@@ -211,14 +215,52 @@ def run(
                 )
                 components.append(output_model_component)
                 component_names.append(component_name)
+                Pass._carry_forward_additional_files(component_model, output_model_component)
             output_model = CompositeModelHandler(components, component_names)
         else:
             output_model = self._run_for_config(model, data_root, config, output_model_path)
+            Pass._carry_forward_additional_files(model, output_model)
+
         # assumption: the model attributes from passes, if any, are more important than
         # the input model attributes, we should not update/extend anymore outside of the pass run
         output_model.model_attributes = output_model.model_attributes or model.model_attributes
         return output_model
 
+    @staticmethod
+    def _carry_forward_additional_files(input_model: OliveModelHandler, output_model: OliveModelHandler):
+        # NOTE: Can't use model.model_path because that always gets resolved to a filepath.
+        # We need the directory path here.
+        input_model_output_path = input_model.get_resource("model_path")
+        if not input_model_output_path:
+            return
+
+        input_model_output_path = Path(input_model_output_path)
+        if not input_model_output_path.is_dir():
+            return
+
+        input_model_attributes = input_model.model_attributes or {}
+        input_model_additional_files = set(input_model_attributes.get("additional_files", []))
+        if not input_model_additional_files:
+            return
+
+        output_model_output_path = Path(output_model.get_resource("model_path"))
+        if not output_model_output_path.is_dir():
```

Comment:
I think this check might fail if the optimization somehow made an onnx model smaller and it was not saved with external data. meaning model path might be file. 

it might be safer to have a different condition for onnx model like:
```python
if not output_model_output_path.is_dir() and isinstance(output_model, ONNXModelHandler):
    # change the "model_path" resource to the parent directory of the model file
    output_model.set_resource("model_path", output_model_output_path.parent)
    output_model.onnx_file_name = output_model_output_path.name
    output_model_output_path = output_model_output_path.parent
```

Reply:
Your solution would pull OnnxModelHandler dependency into Pass. So, had to use the suffix as condition.

Reply:
I think that is fine? We already have dependency on other handler classes. 

Checking for `.onnx` suffix is also fine but in theory there could be an onnx model file without the `.onnx` suffix. This probably won't be the case since most of our passes use the `resolve_onnx_path` util.

Reply:
A base class depending on a subclass is architecturally wrong in a design. It's bad software practice. I would recommend basing the decision on the extension for now unless you strongly feel that "there could potentially be onnx models with a different extension and this check would fail".

Reply:
I don't understand why it is "architecturally wrong" for a base class of one class (Pass) depending on the subclass of another class (ModelHandler). 
Can you point me to some resources where this practice is not recommended?

Reply:
Never mind, I still had the old pass hierarchy in mind where OnnxXXXX pass where direct subclass of Pass. We changed that by making Pass and ModelHandler independent. Pass and OnnxModelHandler are no more in the same class hierarchy. My bad.

I will change that check to refer to the handler directly.

Reply:
Done!